activestorage/lib/active_storage/attached/changes/create_one.rb

```diff
@@ -30,6 +30,7 @@ def upload
 
     def save
       record.public_send("#{name}_attachment=", attachment)
+      record.public_send("#{name}_blob=", blob)
```

Comment:
I'm not sure if this is a bug or not. Realized that the `#{name}_blob` association is not getting properly loaded when attaching a single attachment.

I wrote out a much more detailed description of this in the commit message description itself: https://github.com/rails/rails/commit/139e4b43ab90045d05b71581e6e53b398acb826b

It's a very small change but I added as its own commit since it's not fully related to the core PR (although it was impacting my specs for it). 

I'm happy to remove it if needed. Looking forward to thoughts and feedback here. 

Thanks!

Reply:
Please break this out into its own PR.

Reply:
@georgeclaghorn - Done! Opened https://github.com/rails/rails/pull/35412 and removed the commit from here.